### DISCUSSION TITLE
`history.location.key` equivalent in v6? -- How to check if previous item in history stack exists in react router v6?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [remix-run/react-router#9788] [v5 to v6 Migration] [answered]
In our previous react-router v6 app we had code using `history` like this inside of a route modal:

```tsx
  const handleCancelModal = () => {
   const doesAnyHistoryEntryExist = history.location.key
   if(doesAnyHistoryEntryExist){
      history.goBack()
   }
    else {
      // no entry exists, we landed on the page via direct URL, there is no previous history so fallback to x
       history.push('/some_url);
    }
  };
```

In react router v6, how do we check if there is any entry in the history stack since now there is no `useHistory` hook?

```tsx
  const handleCancelModal = () => {
   const doesAnyHistoryEntryExist =  ?????
   if(doesAnyHistoryEntryExist){
      navigate(-1)
   }
    else {
      // no entry exists, we landed on the page via direct URL, there is no previous history so fallback to x
       navigate('/some_url);
    }
  };
```

**Comment by krisheinrich** [comment, marked as answer]
In v6, you can inspect the `location` object returned by the `useLocation` hook for this use case. When the history stack is empty, the `location.key` will have the value `"default"`.

```javascript
const doesAnyHistoryEntryExist = location.key !== "default";
```

Ref: `history` library [docs](https://github.com/remix-run/history/blob/main/docs/api-reference.md#location.key)

**Reply by cliffordfajardo** [reply]
Thanks! Appreciate the link to the docs 🙏

**Comment by Baterka** [comment]
In my case `location.key` is never `default`

**Comment by davidko5** [comment]
oesn't work as expected if we navigate to a new tab using window.open() or <Link /> with target='_blank'. This way app still doesn't have where to go back but location.key isn't equal to 'default'
